sdk/src/main/scala/com/horizen/account/mempool/MempoolMap.scala

```diff
@@ -101,14 +100,45 @@ class MempoolMap(
     this
   }
 
-  def replaceIfCanPayHigherFee(existingTxId: ModifierId, newTx: SidechainTypes#SCAT, mapOfTxsByNonce: TxIdByNonceMap) = {
+  private[mempool] def addNewTransaction(txByNonceMap: TxIdByNonceMap, ethTransaction: SidechainTypes#SCAT) = {
+    val account = ethTransaction.getFrom
+    val accountSize = getAccountSize(account)
+    if (accountSize + txSizeInSlot(ethTransaction) > MaxSlotsPerAccount) {
+      log.trace(s"Adding transaction $ethTransaction exceeds maximum allowed size per account")
+      throw AccountMemPoolOutOfBoundException(ethTransaction.id)
+    }
+    all.put(ethTransaction.id, ethTransaction)
+    txByNonceMap.put(ethTransaction.getNonce, ethTransaction.id)
+  }
+
+  def getAccountSize(account: SidechainTypes#SCP): Long = {
+    val execSize: Long = executableTxs.get(account).map(executableTxsPerAccount => executableTxsPerAccount.values.foldLeft(0L) { (sum, txId) => sum + txSizeInSlot(all(txId)) }).getOrElse(0L)
```

Comment:
'txSizeInSlot' is based on `tx.size()` method.
every time when executed tx is serialized from to bytes and then returns the `size`.
I think we need to cache `size` value for EthereumTransaction.

Reply:
Done